packages/harmony/src/components/expandable-nav-item/ExpandableNavItem.tsx

```diff
@@ -0,0 +1,114 @@
+import { useMemo, useState } from 'react'
+
+import { useTheme, CSSObject } from '@emotion/react'
+
+import { HarmonyTheme } from '../../foundations/theme'
+import { IconCaretDown, IconCaretRight } from '../../icons'
+import { Flex } from '../layout/Flex'
+import { Text } from '../text'
+
+import type { ExpandableNavItemProps } from './types'
+
+const getStyles = (
```

Comment:
im into this! typically we just create the constants in the file. does the memoization make a big deal from what you can tell?

Reply:
was more of a premature optimization to reduce rerenders, navbar seems more snappy after adding it, but not sure if that was my computer or the `useMemo` haha